omnigibson/scenes/scene_base.py

```diff
@@ -169,85 +204,137 @@ def object_registry_group_keys(self):
-        try:
-            task_metadata = scene_info["metadata"]["task"]
-        except:
-            pass
+        # Add the prebuilt scene USD to the stage
+        scene_relative_path = f"/scene_{self.idx}"
+        scene_absolute_path = f"/World{scene_relative_path}"
+
+        # If there is already a prim at the absolute path, the scene has been loaded. If not, load the prebuilt scene USD now.
+        if self.scene_file is not None:
+            scene_prim_obj = og.sim.stage.GetPrimAtPath(scene_absolute_path)
+            if not scene_prim_obj:
+                scene_prim_obj = add_asset_to_stage(asset_path=self.prebuild(), prim_path=scene_absolute_path)
+
+        # Store world prim and load the scene into the simulator
+        self._scene_prim = XFormPrim(
+            relative_prim_path=scene_relative_path,
+            name=f"scene_{self.idx}",
+            load_config={"created_manually": True},
+        )
+        self._scene_prim.load(None)
+        if self.scene_file is not None:
+            assert self._scene_prim.prim_path == scene_prim_obj.GetPath().pathString, "Scene prim path mismatch!"
+
+        # Go through and load all systems.
+        self._load_systems()
 
         # Create desired systems
-        for system_name in init_systems:
+        for system_name in self._init_systems:
             if gm.USE_GPU_DYNAMICS:
-                get_system(system_name)
+                self.get_system(system_name)
             else:
                 log.warning(f"System {system_name} is not supported without GPU dynamics! Skipping...")
 
-        # Iterate over all scene info, and instantiate object classes linked to the objects found on the stage
-        # accordingly
-        for obj_name, obj_info in init_info.items():
-            # Check whether we should load the object or not
-            if not self._should_load_object(obj_info=obj_info, task_metadata=task_metadata):
-                continue
-            # Create object class instance
-            obj = create_object_from_init_info(obj_info)
+        # Position the scene prim initially at a z offset to avoid collision
+        self._scene_prim.set_position([0, 0, m.INITIAL_SCENE_PRIM_Z_OFFSET if self.idx != 0 else 0])
+
+        # Now load the objects with their own logic
+        for obj_name, obj in self._init_objs.items():
             # Import into the simulator
-            og.sim.import_object(obj)
+            self.add_object(obj)
             # Set the init pose accordingly
-            obj.set_position_orientation(
-                position=init_state[obj_name]["root_link"]["pos"],
-                orientation=init_state[obj_name]["root_link"]["ori"],
+            obj.set_local_pose(
+                position=self._init_state[obj_name]["root_link"]["pos"],
+                orientation=self._init_state[obj_name]["root_link"]["ori"],
             )
 
+        # Position the scene prim based on the last scene's right edge
+        global LAST_RIGHT_EDGE_POS
```

Comment:
off not sure I like this global variable, why not moving this logic to og.sim, i.e. og.sim keeps track of all the scenes it has imported so far?